docs/design/persistent-storage.md

```diff
@@ -0,0 +1,260 @@
+```
+
+A user requests persistent storage by creating a `PersistentVolumeClaim`.  This claim is namespaced scoped and is bound to the volume when the request is fulfilled.
+
+The pod author attaches the claim to her pod's volume.  The scheduler and the volume's controller ensure the pod and volume are collocated on the host.
+
+
+
+#### Pod to specify storage requirements
+
+`PersistenceSpec` are the capabilities and performance characteristics a pod author can request and rely on for the entire lifecycle of the volume.  Requests for storage that do not match available storage will go unfulfilled.
+
+```
+	PersistenceSpec
+			Mode				ReadWriteOnce | ReadOnlyMany
+			RetentionPolicy 	VolumeRetentionPolicy (RetainOnDelete|RecycleOnDelete)			Resources			ResourcesList (size, iops, throughput)
+			Selector			Labels to identify the PersistentVolume backing implementation
+```
+
+
+
+#### Storage discovery 
+
+An API and CLI is needed to view available and in-use volumes by namespace.
+
+**Use Case**:  Susie needs to crunch some data and serve it afterwards.  A single pod to create data is sufficient, but she estimates needing 10 web front ends to serve the data afterwards.  For this use case, the data does not change once created.  Susie first looks to see what is available in the cluster.
+
+```
+$ cluster/kubectl.sh get storage
+
+NAME                LABELS              MODES               RESOURCES
+_pv_vol_01          name=my-data        RWO,ROX             size=5gb,iops=30
+_pv_vol_02                              RWO,ROX             size=5gb,iops=30
+_pv_vol_03          name=other-data     RWO,ROX             size=25gb,iops=30
+
+```
+
+#### Request persistent storage for a pod
+
+Susie is happy to see storage that can be mounted many times (ReadOnlyMany).  That satisfies her load balancer requirement.  Susie creates an pod image with her number crunching code and requests persistence for her volume.
+
+Susie's YAML looks something like this for the pod that creates her data:
+
+
+```yaml
+
+id: creator-of-data
+kind: Pod
+namespace: ns
+apiVersion: v1beta1
+desiredState:
+  manifest:
+    version: v1beta1
+    id: creator-of-data
+    containers:
+      - name: numbercruncher
+        image: ns/makesomdata:v1
+        command:
+          - /run.sh
+        volumeMounts:
+          - name: data
+            mountPath: /some_data
+    volumes:
+      - name: data
+        PersistenceSpec:
+        	availableModes: readwriteonce | readonlymany
+        	currentMode: readwriteonce
+        	retentionPolicy: retainondelete
+        	resources:
+        		- size: 5
+        	selector:
+        		- name: my-persistent-data
+labels:
+  name: creator-of-data
+
+```
+
+#### PersistentVolumeBinder
+
+`PersistentVolumeBinder` is running in master, watching for new `PersistentVolumeClaims` and matching them to a volume from the pool.  
+
+``` 
+	// bind a volume with a claim 
+	PersistentVolume.Status.ClaimantRef = PersistentVolumeClaim object reference
```

Comment:
It seems like this should be `ClaimRef`, not `ClaimantRef` --  the claimant would be the person or thing making the claim, not the claim itself.